libs/@blockprotocol/graph/src/codegen/entity-type-to-typescript.ts

```diff
@@ -218,8 +217,9 @@ const _jsonSchemaToTypeScript = async (
       generateLinkEntityAndRightEntityDefinition({
         sourceEntityTypeName: typeName,
         linkEntityTypeName: linkEntityType.typeName,
-        rightEntityTypeNames: (
-          rightEntityTypes ?? [{ typeName: "Entity" }]
+        rightEntityTypeNames: (rightEntityTypes.length > 0
+          ? rightEntityTypes
+          : [{ typeName: "Entity" }]
```

Comment:
Bug fix: `rightEntityTypes` is an empty array if there is no constraint on the expected type